examples/coap-server.c

```diff
@@ -241,28 +241,38 @@ hnd_get_index(coap_resource_t *resource,
 }
 
 static void
-hnd_get_time(coap_resource_t *resource,
-             coap_session_t *session,
-             const coap_pdu_t *request,
-             const coap_string_t *query,
-             coap_pdu_t *response) {
+hnd_get_fetch_time(coap_resource_t *resource,
+                   coap_session_t *session,
+                   const coap_pdu_t *request,
+                   const coap_string_t *query,
+                   coap_pdu_t *response) {
   unsigned char buf[40];
   size_t len;
   time_t now;
   coap_tick_t t;
   (void)request;
+  coap_pdu_code_t code = coap_pdu_get_code(request);
+  size_t size;
+  const uint8_t *data;
 
   if (my_clock_base) {
 
     /* calculate current time */
     coap_ticks(&t);
     now = my_clock_base + (t / COAP_TICKS_PER_SECOND);
 
-    if (query != NULL
-        && coap_string_equal(query, coap_make_str_const("ticks"))) {
-          /* output ticks */
-          len = snprintf((char *)buf, sizeof(buf), "%u", (unsigned int)now);
+    /* coap_get_data() sets size to 0 on error */
+    (void)coap_get_data(request, &size, &data);
 
+    if (code == COAP_REQUEST_CODE_GET && query != NULL &&
+        coap_string_equal(query, coap_make_str_const("ticks"))) {
```

Comment:
I suggest we create the "ticks"-string before the conditional and use `ticks->s`, `ticks->length` also in the `memcmp()` for `COAP_REQUEST_CODE_FETCH`. This could make later changes less error prone.

Reply:
OK.  Taken this principle and updated code pushed.